src/assembly/manifest.ts

```diff
@@ -0,0 +1,166 @@
+                const stackArtifact = this.assembly.getStackArtifact(artifactId);
+                this.stacks.set(artifactId, stackArtifact);
+                const metadata: StackMetadata = {};
+                const props = artifact.properties as AwsCloudFormationStackProperties;
+                const template: CloudFormationTemplate = fs.readJSONSync(
+                    path.resolve(this.directory, props.templateFile),
+                );
+                for (const [metadataId, metadataEntry] of Object.entries(artifact.metadata ?? {})) {
+                    metadataEntry.forEach((meta) => {
+                        if (meta.type === ArtifactMetadataEntryType.LOGICAL_ID) {
+                            // For some reason the metadata entry prefixes the path with a `/`
+                            const path = metadataId.startsWith('/') ? metadataId.substring(1) : metadataId;
+                            metadata[path] = meta.data as LogicalIdMetadataEntry;
+                        }
+                    });
+                }
+                const assets = this.getAssetsForStack(artifactId);
+                const stackTree = this.tree.children![artifactId];
+                const stackManifest = new StackManifest(
+                    this.directory,
+                    artifactId,
+                    props.templateFile,
+                    metadata,
+                    stackTree,
+                    template,
+                    assets,
+                );
+                this._stackManifests.set(artifactId, stackManifest);
+            }
+        }
+    }
+
+    /**
+     * Get the stacks from the Cloud Assembly
+     *
+     * @returns List of CloudFormationStackArtifacts available in the Cloud Assembly
+     */
+    public get stackManifests(): StackManifest[] {
+        return Array.from(this._stackManifests.values());
+    }
+
+    /**
+     * Return a list of assets for a given stack
+     *
+     * @param stackId - The artifactId of the stack to find assets for
+     * @returns a list of `StackAsset` for the given stack
+     */
+    private getAssetsForStack(stackId: string): StackAsset[] {
+        const assets: (FileManifestEntry | DockerImageManifestEntry)[] = [];
+        for (const artifact of Object.values(this.manifest.artifacts ?? {})) {
+            if (
+                artifact.type === ArtifactType.ASSET_MANIFEST &&
+                (artifact.properties as AssetManifestProperties)?.file === `${stackId}.assets.json`
+            ) {
+                assets.push(...this.assetsFromAssetManifest(artifact));
+            }
+        }
+        return assets;
+    }
+
+    /**
+     * Get a list of assets from the asset manifest.
+     *
+     * @param artifact - An ArtifactManifest to extract individual assets from
+     * @returns a list of file and docker assets found in the manifest
+     */
+    private assetsFromAssetManifest(artifact: ArtifactManifest): StackAsset[] {
```

Comment:
Is this static? Trying to find if it uses `this` and it does not seem to.

Reply:
It uses `this.directory`